src/offset/local/tz_info/timezone.rs

```diff
@@ -933,6 +936,338 @@ mod tests {
+            NaiveTime::from_hms_opt(2, 0, 0).unwrap(),
+        );
+        let offsets = tz_ref.find_local_time_type_from_local(date.and_utc().timestamp(), 2024)?;
+        // The earlier time is the time still in DST.
+        assert_eq!(offsets.earliest().unwrap().is_dst, true);
+        assert_eq!(offsets.earliest().unwrap().ut_offset, 7200);
+        assert_eq!(offsets.earliest().unwrap().name.unwrap().as_bytes(), b"CEST");
+        // The later time is the standard time.
+        assert_eq!(offsets.latest().unwrap().is_dst, false);
+        assert_eq!(offsets.latest().unwrap().ut_offset, 3600);
+        assert_eq!(offsets.latest().unwrap().name.unwrap().as_bytes(), b"CET");
+
+        // Southern Hemisphere (Chili)
+        let tz: TimeZone = TimeZone::from_posix_tz("<-04>4<-03>,M9.1.6/24,M4.1.6/24")?;
+        let tz_ref = tz.as_ref();
+        // For 2024 DST ends at 0:00 on April 7, moving back 1 hour.
+        let date = NaiveDateTime::new(
+            NaiveDate::from_ymd_opt(2024, 4, 7).unwrap(),
+            NaiveTime::from_hms_opt(0, 0, 0).unwrap(),
+        );
+        let offsets = tz_ref.find_local_time_type_from_local(date.and_utc().timestamp(), 2024)?;
+        // The earlier time is the time still in DST.
+        assert_eq!(offsets.earliest().unwrap().is_dst, true);
+        assert_eq!(offsets.earliest().unwrap().ut_offset, -10800);
+        assert_eq!(offsets.earliest().unwrap().name.unwrap().as_bytes(), b"-03");
+        // The later time is the standard time.
+        assert_eq!(offsets.latest().unwrap().is_dst, false);
+        assert_eq!(offsets.latest().unwrap().ut_offset, -14400);
+        assert_eq!(offsets.latest().unwrap().name.unwrap().as_bytes(), b"-04");
+
+        // Reverse DST (Europe/Dublin)
+        let tz: TimeZone = TimeZone::from_posix_tz("IST-1GMT0,M10.5.0,M3.5.0/1")?;
+        let tz_ref = tz.as_ref();
+        // For 2024 DST *starts* at 2:00 on October 27, moving back 1 hour.
+        let date = NaiveDateTime::new(
+            NaiveDate::from_ymd_opt(2024, 10, 27).unwrap(),
+            NaiveTime::from_hms_opt(2, 0, 0).unwrap(),
+        );
+        let offsets = tz_ref.find_local_time_type_from_local(date.and_utc().timestamp(), 2024)?;
+        // The earlier time is the time still in DST.
+        assert_eq!(offsets.earliest().unwrap().is_dst, false);
+        assert_eq!(offsets.earliest().unwrap().ut_offset, 3600);
+        assert_eq!(offsets.earliest().unwrap().name.unwrap().as_bytes(), b"IST");
+        // The later time is the standard time.
+        assert_eq!(offsets.latest().unwrap().is_dst, true);
+        assert_eq!(offsets.latest().unwrap().ut_offset, 0);
+        assert_eq!(offsets.latest().unwrap().name.unwrap().as_bytes(), b"GMT");
+
+        Ok(())
+    }
+
+    #[test]
+    #[allow(clippy::bool_assert_comparison)]
+    fn test_dst_backward_tzfile() -> Result<(), Error> {
+        // Northern hemisphere DST (CET/CEST)
+        let data: [u8; 604] = [
```

Comment:
These big binary blobs are a pain to manage. What value does this test have over and above the `test_dst_backward_posix_tz` tests? Do these tests actually gain us any coverage?